src/prefect/runner.py

```diff
@@ -0,0 +1,853 @@
+        self._limiter: Optional[anyio.CapacityLimiter] = (
+            anyio.CapacityLimiter(limit) if limit is not None else None
+        )
+        self._client = get_client()
+        self._submitting_flow_run_ids = set()
+        self._cancelling_flow_run_ids = set()
+        self._scheduled_task_scopes = set()
+        self._deployment_ids: Set[UUID] = set()
+        self._flow_run_process_map = dict()
+
+    @sync_compatible
+    async def add_deployment(
+        self,
+        deployment: RunnerDeployment,
+    ) -> UUID:
+        """
+        Registers the deployment with the Prefect API and will monitor for work once
+        the runner is started.
+
+        Args:
+            deployment: A deployment for the runner to register.
+        """
+        deployment_id = await deployment.apply()
+        self._deployment_ids.add(deployment_id)
+
+        return deployment_id
+
+    @sync_compatible
+    async def add(
+        self,
+        flow: Flow,
+        name: str = None,
+        interval: Optional[Union[int, float, datetime.timedelta]] = None,
+        cron: Optional[str] = None,
+        rrule: Optional[str] = None,
+        parameters: Optional[dict] = None,
+        triggers: Optional[List[DeploymentTrigger]] = None,
+        description: Optional[str] = None,
+        tags: Optional[List[str]] = None,
+        version: Optional[str] = None,
+    ) -> UUID:
+        """
+        Provides a flow to the runner to be run base on the provided configuration.
+
+        Will create a deployment for the provided flow and register the deployment
+        with the runner.
+
+        Args:
+            flow: A flow for the runner to run.
+            name: The name to give the created deployment. Will default to the name
+                of the runner.
+            interval: An interval on which to execute the current flow. Accepts either a number
+                or a timedelta object. If a number is given, it will be interpreted as seconds.
+            cron: A cron schedule of when to execute runs of this flow.
+            rrule: An rrule schedule of when to execute runs of this flow.
+            triggers: A list of triggers that should kick of a run of this flow.
+            parameters: A dictionary of default parameter values to pass to runs of this flow.
+            description: A description for the created deployment. Defaults to the flow's
+                description if not provided.
+            tags: A list of tags to associate with the created deployment for organizational
+                purposes.
+            version: A version for the created deployment. Defaults to the flow's version.
+        """
+        # TODO: expose a filesystem interface with hot reloading
+        # will need to create a separate method for deployment creation
+        api = PREFECT_API_URL.value()
+        if any([interval, cron, rrule]) and not api:
+            self._logger.warning(
```

Comment:
Good question - if you specify a schedule but are not either running against Cloud, or if you forgot to run `prefect server start`, then there is no scheduler actively scheduling your work; the warning is trying to catch that early to prevent confusion.